### PR TITLE
Add instructions for debugging ARM with VSCode+BMP.

### PR DESCRIPTION
**Pull request by tzarc**
## Description

Adds some documentation to the VSCode documentation page to describe a method do debug ARM MCUs using cortex-debug and a Black Magic Probe.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/other_vscode.md [hunk 1]
```diff
@@ -167,7 +167,7 @@ You'll need to perform some modifications to the file above in order to target y
 * `"BMPGDBSerialPort"`: The `COM` port under Windows, or the `/dev/...` path for Linux/macOS.
 * `"svdFile"`: _[Optional]_ The path to the SVD file that defines the register layout for the MCU -- the appropriate file can be downloaded from the [cmsis-svd repository](https://github.com/posborne/cmsis-svd/tree/master/data/STMicro)
 * `"device"`: The name of the MCU, which matches the `<name>` tag at the top of the downloaded `svd` file.
-* `"armToolchainPath"`: _[Optional]_ The path to the ARM toolchain installation location on Windows -- under normal circumstances Linux/macOS will auto-detect this correctly and will not need to be specified. Windows is special, which is why it has its own section.
+* `"armToolchainPath"`: _[Optional]_ The path to the ARM toolchain installation location on Windows -- under normal circumstances Linux/macOS will auto-detect this correctly and will not need to be specified. Windows is special, which is why it has its own section. You can get a copy of the GNU ARM toolchain for Windows from [developer.arm.com](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads).
```
Comment (fauxpark) [inline docs/other_vscode.md:170]: MSYS2 has an arm-none-eabi-gdb package, just FYI

Comment (tzarc) [reply docs/other_vscode.md:170]: Does it need to be executed from within the MSYS2 environment? VSCode executes externally... unsure if it'll work.

Comment (fauxpark) [reply docs/other_vscode.md:170]: The entire set of ARM toolchain packages are compiled as native binaries (this is the `mingw-w64-x86_64` or `i686` prefix), so they don't depend on [msys-2.0.dll](https://www.msys2.org/wiki/How-does-MSYS2-differ-from-Cygwin/), unlike eg. pacman and git. Even so, I seem to be able to run MSYS packages from the command prompt just fine too.

Comment (tzarc) [reply docs/other_vscode.md:170]: Interesting... guess I should set up a Windows env for QMK and try it out. Should simplify the docs significantly if we can leverage it, especially considering it'll match the compiler build the firmwares get built with.

Comment (drashna) [reply docs/other_vscode.md:170]: For macOS, actually, the gdb binary isn't present, and has to be downloaded manually.  

Though, correct me if I'm wrong, but that is something that we can fix/add, I think.

Comment (fauxpark) [reply docs/other_vscode.md:170]: There is `arm-none-eabi-gdb` in the core tap.

Comment (drashna) [reply docs/other_vscode.md:170]: Ah, last I cheched, I swear it wasn't.  Also, it's 13.2, and not sure if the version matching gcc matters here.

File: docs/other_vscode.md [hunk 2]
```diff
@@ -117,3 +117,77 @@ Using the [standard `compile_commands.json` database](https://clang.llvm.org/doc
...
+  ]
+}
+```
+
+You'll need to perform some modifications to the file above in order to target your specific device:
+
+* `"name"`: Can be anything, but if you're debugging multiple targets you'll want something descriptive here.
+* `"cwd"`: The path to the QMK Firmware repository root directory -- _if using the `.vscode` directory existing in the `qmk_firmware` git repository, the default above should be correct_
+* `"executable"`: The path to the `elf` file generated as part of the build for your keyboard -- _exists in `<qmk_firmware>/.build`_
+* `"BMPGDBSerialPort"`: The `COM` port under Windows, or the `/dev/...` path for Linux/macOS.
```
Comment (drashna) [inline docs/other_vscode.md:171]: messing with this, it seems that it creates 2 serial ports, for me.  Might be worth mentioning that, and to use the first?

**Comment by drashna** [comment]
Aside from that, tested this out and it works well.  

Though, would it worth be adding a section for adding tasks to vscode, especially for the prelaunchtask?  Not here necessarily, but just in general.

File: docs/other_vscode.md (hunk 1 above)
Comment (tzarc) [reply docs/other_vscode.md]: Added notes to point MSYS2 at `gdb-multiarch.exe`. Rest should be sweet, now.

File: docs/other_vscode.md (hunk 2 above)
Comment (tzarc) [reply docs/other_vscode.md]: Done.

**Review by KarlK90 — COMMENTED** [review]
LGTM, just a little note on the compiler flags

File: docs/other_vscode.md [hunk 3]
```diff
@@ -117,3 +117,77 @@ Using the [standard `compile_commands.json` database](https://clang.llvm.org/doc
...
+* `"armToolchainPath"`: _[Optional]_ The path to the ARM toolchain installation location on Windows -- under normal circumstances Linux/macOS will auto-detect this correctly and will not need to be specified. 
+
+!> Windows builds of QMK Firmware are generally compiled using QMK MSYS, and the path to gdb's location (`C:\\QMK_MSYS\\mingw64\\bin`) needs to be specified under `armToolchainPath` for it to be detected. You may also need to change the GDB path to point at `C:\\QMK_MSYS\\mingw64\\bin\\gdb-multiarch.exe` in the VSCode Cortex-Debug user settings: ![VSCode Settings](https://i.imgur.com/EGrPM1L.png)
+
+Optionally, the following modifications should also be made to the keyboard's `rules.mk` file to disable optimisations -- not strictly required but will ensure breakpoints and variable viewing works correctly:
+```makefile
+# Disable optimisations for debugging purposes
+LTO_ENABLE = no
+OPT = 0
+OPT_DEFS += -g
```
[the remark this answers is not in the document]
Comment (tzarc) [reply docs/other_vscode.md:181]: For me, no real preference; if you want to make a suggestion on the PR for what it _should_ look like I'll be more than happy to commit it.

Comment (KarlK90) [inline docs/other_vscode.md:181]:
```suggestion
OPT = g
DEBUG = 3
```

Sure, that would be my suggestion :-).

**Review by drashna — APPROVED** [review]

**Pull request merged**